.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,304 @@
+name: Fetch All Open and Recently Closed Pull Requests using cron job
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 15 * * 1-5'  # At 15:30 on every day-of-week from Monday through Friday(UTC Time), 9pm in IST and 11:30 AM in EST.
+
+jobs:
+  fetch_all_pull_requests_and_notify_using_condition_and_cron:
+    runs-on: ubuntu-latest
+
+    env:
+      REPO: ${{ github.repository }}
+      BRANCH: main
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      API_URL: https://api.github.com/repos/redhat-developer/lsp4ij/pulls
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Fetch recent cache
+        run: |
+          gh extension install actions/gh-actions-cache
+
+          # Fetching list of cache keys........
+          cacheKeys=$(gh actions-cache list -R $REPO -B $BRANCH -L 100 | cut -f 1 )
+          
+          # Extract the most recent cache key
+          mostRecentCacheKey=$(echo "$cacheKeys" | head -n 1)
+          
+          if [ -n "$mostRecentCacheKey" ]; then
+            echo "CACHE_KEY=$mostRecentCacheKey" >> $GITHUB_ENV
+          else
+            echo "No recent cache key found, generating a new one...."
+            echo "CACHE_KEY=${{ runner.os }}-pr-cache-${{ github.run_number }}" >> $GITHUB_ENV
+          fi
+          
+          echo "Done"
+          # 'secrets.GITHUB_TOKEN' is automatically provided by GitHub Actions for each workflow run. You don't need to manually create or manage this token.
+
+      - name: Restore cache
+        id: cache-restore
+        uses: actions/cache@v2
+        with:
+          path: cache
+          key: ${{ env.CACHE_KEY }}
+
+      - name: Ensure cache directory exists and check if cache restored
+        run: |
+          mkdir -p cache
+          if [ -f cache/notified_prs.json ]; then
+            echo "Cache restored successfully."
+            cat cache/notified_prs.json
```

Comment:
Do we need this cat command here?

Reply:
This is just for printing to ensure that the cache was fetched correctly, I hope this will help in the future to check the fetched data correctly. If required, we can delete it.

<img width="654" alt="Screenshot 2024-06-25 at 2 17 40 PM" src="https://github.com/OpenLiberty/liberty-tools-intellij/assets/133869023/0d6a54d2-fe8e-43ac-9f8b-74d6409774fe">